src/classes/CRLP_RollupUI_SVC.cls

```diff
@@ -906,6 +908,49 @@ public with sharing class CRLP_RollupUI_SVC {
         return fieldObjectList;
     }
 
+    /*******************************************************************************************************
+    * @description Checks that a given field is allowed as a target field
+    * @param objectName the name of the object
+    * @param dfr the DescribeFieldResult of the field
+    * @return Boolean
+    */
+    public static Boolean isTargetFieldAllowed(String objectName, Schema.DescribeFieldResult dfr) {
+
+        if(!dfr.isUpdateable()
+                || !dfr.isCustom()
+                || dfr.name.toLowerCase().contains('__system')
+                || dfr.name.contains('Latitude__s')
+                || dfr.name.contains('Longitude__s')) {
+            return false;
+        }
+
+        Map<String,Set<String>> bannedFieldsByObject = new Map<String,Set<String>>();
+
+        // BAN ALL NPSP FIELDS ON RD EXCEPT THE FOUR CORE ROLLUP FIELDS
+        bannedFieldsByObject.put('npe03__Recurring_Donation__c',new Set<String>{'npe03__Amount__c'
+                ,'npe03__Recurring_Donation_Campaign__c'
+                ,'npe03__Contact__c'
+                ,'npe03__Date_Established__c'
+                ,'npe03__Donor_Name__c'
+                ,'npe03__Installment_Amount__c'
+                ,'npe03__Installment_Period__c'
+                ,'npe03__Installments__c'
+                ,'npe03__Open_Ended_Status__c'
+                ,'npe03__Organization__c'
+                ,'npe03__Schedule_Type__c'
+                ,'npe03__Total__c'
+                ,'Always_Use_Last_Day_Of_Month__c'});
```

Comment:
This last field is in the Cumulus repo, so it will get the `npsp__` namespace when delivered to customers. It'll need a call to the UTIL_Namespace class here to apply the namespace if required.